src/lib/components/Uploader/uploader.component.js

```diff
@@ -0,0 +1,236 @@
+            throw error;
+          }
+          // Get destination file url
+          const destinationUri = `${fileBase}/${encodeURIComponent(
+            file.name
+          )}${suffix}`;
+
+          // Send file on Base64 to server using fetch from solid-auth-client
+          const response = await auth.fetch(destinationUri, {
+            method: "PUT",
+            force: true,
+            headers: {
+              "content-type": file.type,
+              credentials: "include"
+            },
+            body: data
+          });
+          // Is all is fine we add new files into uploadedFiles array
+          if (response.ok) {
+            const newUploadedFiles = [
+              ...this.state.uploadedFiles,
+              { uri: destinationUri, name: file.name }
+            ];
+
+            return this.setState({ uploadedFiles: newUploadedFiles });
+          }
+          // If something come bad throw error
+          throw response;
+        } catch (error) {
+          onError(error, file);
+        }
+      };
+      reader.readAsArrayBuffer(file);
+    });
+  };
+  /**
+   * will fire when all files was uploaded
+   * @params {Array<UpoadFiles>} uploadFiles
+   */
+  onComplete = (uploadedFiles: Array<UploadedFiles>) => {
+    if (this.state.uploadedFiles.length === this.state.files.length) {
+      this.setState({ inProgress: false });
+
+      if (this.props.onComplete) {
+        this.props.onComplete(uploadedFiles);
+      }
+    }
+  };
+  onDragEnter = (event: React.DragEvent) => {
+    this.overrideEvent(event);
+    // Counter drag events
+    this.counter += 1;
+
+    if (
+      (event.dataTransfer.items &&
+        event.dataTransfer.items[this.positionFile]) ||
+      (event.dataTransfer.types &&
+        event.dataTransfer.types[this.positionFile] === "Files")
+    ) {
+      this.setState({ dragging: true });
+    }
+  };
+  onDragLeave = (event: React.DragEvent) => {
+    this.overrideEvent(event);
+    this.counter -= 1;
+
+    if (this.counter === 0) {
+      this.setState({ dragging: false });
+    }
+  };
+  onDrop = async (event: React.DragEvent) => {
+    this.overrideEvent(event);
+    this.counter = 0;
+
+    let files = [];
+
+    if (
+      this.props.limitFiles &&
+      event.dataTransfer.items.length > this.props.limitFiles
+    ) {
+      const error = {
+        type: "file",
+        statusText: "Sorry you exceed files allowed per upload",
+        code: 400
+      };
+
+      return this.props.onError(error, []);
+    }
+
+    if (event.dataTransfer.items) {
+      files = event.dataTransfer.files.map(file => file);
+    }
+
+    if (this.props.onDrop) {
+      this.props.onDrop(files);
+    }
+
+    this.onStart();
+    this.setState({ files });
+  };
+  onClickFile = () => {
+    if (this.fileInput) {
+      this.fileInput.current.click();
+    }
+  };
+  /**
+   * Will call when file start to upload.
+   */
+  onStart = () => {
+    // If onStart callback come will fire it.
```

Comment:
I'm not sure what this comment means

Reply:
it's: When upload start this event will fire